### ISSUE TITLE
TODO: Examples page

### ISSUE THREAD

**Issue by caffeine-addictt** [python-thread/thread.ngjx.org#15]
# Feature Request
Your issue may already be reported!
Please check out our [active issues](https://github.com/python-thread/thread.ngjx.org/issues) before creating one.

## Is Your Feature Request Related to an Issue?

Thread may appear overly confusing to use, especially for those unfamiliar with python's built-in threading library.
At present, there is no examples for usage of `thread`, providing these examples can help guide first-time developers.

## Describe the Solution You'd Like

A `'/examples' or equivalent` route should be created to provide easy-to-follow and heavily documented examples for using `thread`.

**Comment by ShiroTohu** [comment]
Just a thought that I had in passing. Instead of an `examples/` route we could implement a `learn/` route instead that covers the basics of the python thread module and the thread module. Next.JS utilizes this same approach and I quite liked it when I was learning Next.JS for the website. What's also nice about it is that it covers React as well which reduced the knowledge of entry! so I think it could work wonderfully here as well.

See what I am talking about [here](https://nextjs.org/learn?utm_source=next-site&utm_medium=homepage-cta&utm_campaign=home).

I could also add a quiz component like seen in Next.JS course!

> A `'/examples' or equivalent` route should be created to provide easy-to-follow and heavily documented examples for using thread.

We can probably cover more heavy concepts in the actual documentation for people more familiar with the base library and the utilities thread provides. We could probably put it under a sub-folder called examples or pair it with the relative methods, whichever is more convenient for reader.

What do you think?

**Comment by caffeine-addictt** [comment]
> Just a thought that I had in passing. Instead of an `examples/` route we could implement a `learn/` route instead that covers the basics of the python thread module and the thread module. Next.JS utilizes this same approach and I quite liked it when I was learning Next.JS for the website. What's also nice about it is that it covers React as well which reduced the knowledge of entry! so I think it could work wonderfully here as well.

Ah! I didn't think of that. Yes, I agree this would be a better approach. I'll update the initial issue.


> We can probably cover more heavy concepts in the actual documentation for people more familiar with the base library and the utilities thread provides. We could probably put it under a sub-folder called examples or pair it with the relative methods, whichever is more convenient for reader.

I guess it would make more sense to keep the examples simple. We could instead provide links to the more in-depth documentation.

Perhaps something of this structure?
```
/
  -> /about
  -> ...
  -> /learn
      -> /[slug] (Links to [/slug])
  -> /[slug]
```

**Comment by ShiroTohu** [comment]
> I guess it would make more sense to keep the examples simple. We could instead provide links to the more in-depth documentation.

Yep! I think that idea. As for the structure you mentioned, take a look at the example pages below.

![Example in the tutorial](https://github.com/python-thread/thread.ngjx.org/assets/44159210/20342257-afdb-40bb-b880-6543ea48cb1a)

---

![More in-depth example](https://github.com/python-thread/thread.ngjx.org/assets/44159210/b9844fcb-05ef-44ec-a48c-b99e4f16381f)

> [!IMPORTANT]  
> Just want to point out that the documentation will not look like the second picture of course. Just a placeholder :)

**Comment by caffeine-addictt** [comment]
Yea! I believe we are on the same page 👍🏻

**Comment by caffeine-addictt** [comment]
Which library are you using for syntax highlighting? `react-syntax-highlighter/prisma`?

**Comment by ShiroTohu** [comment]
> Which library are you using for syntax highlighting? `react-syntax-highlighter/prisma`?

I'm unsure how to find out.

I haven't touched anything pertaining to syntax highlighting on my end. So I don't really know what to look for.

**Comment by caffeine-addictt** [comment]
Ah!! I see.

**Comment by caffeine-addictt** [comment]
I'm thinking we should still include a `/learn/examples` route, but just have it be for getting a quick-start.

This way, we could encapsulate all the examples and learning within `/learn` without a `/getting-started` etc.

Something like this perhaps?
```
/
  -> ...
  -> /learn
      -> /examples
          -> /[slug]
      -> /[slug] (Links to [/slug])
  -> /[slug]

```

**Comment by caffeine-addictt** [comment]
I'll go ahead and abstract the cautions to its own `/learn/cautions` page.
